src/ln/channelmonitor.rs

```diff
@@ -97,7 +99,11 @@ pub trait ManyChannelMonitor: Send + Sync {
 	/// Implementor must also ensure that the funding_txo outpoint is registered with any relevant
 	/// ChainWatchInterfaces such that the provided monitor receives block_connected callbacks with
 	/// any spends of it.
-	fn add_update_monitor(&self, funding_txo: OutPoint, monitor: ChannelMonitor) -> Result<(), ChannelMonitorUpdateErr>;
+	fn add_update_monitor(&self, funding_txo: OutPoint, monitor: ChannelMonitor, pending_htlc_sources: Vec<(PaymentHash, (OutPoint, HTLCSource))>) -> Result<(), ChannelMonitorUpdateErr>;
```

Comment:
Same here, I thought we weren't doing the pending_htlc_sources change.

Reply:
Refactored